packages/js-moi-providers/src/base-provider.ts

```diff
@@ -358,6 +395,120 @@ export class BaseProvider extends AbstractProvider {
+
+            const response: RpcResponse = await this.execute("moi.NewTesseractsByAccountFilter", params);
+
+            return this.processResponse(response);
+        } catch (error) {
+            throw error;   
+        }
+    }
+
+    /**
+     * Initiates a filtering mechanism to fetch recently identified pending interaction.
+     * The filter setup triggers a 1-minute timeout period, and with each subsequent request,
+     * the timeout is reset to 1 minute.
+     * 
+     * @returns {Promise<Filter>} A object containing the Filter ID for PendingIxnsFilter
+     * @throws {Error} Throws an error if there is an error executing the RPC call.
+     */
+    public async getPendingInteractionFilter(): Promise<Filter> {
+        try {
+            const params = null;
+
+            const response: RpcResponse = await this.execute('moi.PendingIxnsFilter', params);
+            
+            return this.processResponse(response);
+        } catch (error) {
+            throw error;
+        }
+    }
+
+    /**
+     * Asynchronously removes the filter and returns a Promise that resolves to a
+     * object.
+     * The object has a `status` property, which is true if the filter is successfully removed, otherwise false.
+     * 
+     * @returns {Promise<FilterDeletionResult>} A Promise that resolves to an object with a `status` property indicating the success of the filter removal.
+     * @throws {Error} Throws an error if there is an error executing the RPC call.
+     */
+    public async removeFilter(filter: Filter): Promise<FilterDeletionResult> {
+        try {
+            const params = {
+                id: filter.id
+            };
+
+            const response: RpcResponse = await this.execute("moi.RemoveFilter", params);
+
+            return this.processResponse(response);
+        } catch (error) {
+            throw error;
+        }
+    }
+
+    
+    /**
+     * Retrieves all filter changes since the last poll.
+     *
+     * The specific result varies depending on the type of filter used.
+     * 
+     * @param {Filter} filter - The filter object for which changes are to be retrieved.
+     *
+     * @returns {Promise<T>} A promise that resolves to an object containing information about the changes made to the specified filter since the last poll. The structure of the object is determined by the type of filter provided.
+     * @throws {Error} Throws an error if there is an issue executing the RPC call.
+     * 
+     * @template T - The type of the object returned, dependent on the provided filter.
+     */
+    async getFilterChanges<T extends any>(filter: Filter): Promise<T> {
+        try {
+            const params = {
+                id: filter.id
+            };
+
+            const response = await this.execute("moi.GetFilterChanges", params);
+
+            if(response.result.data === null) {
+                return response.result.data;
+            }
+
+            return this.processResponse(response);
```

Comment:
it would be better to make the necessary changes in the process response method.

Reply:
e26dfd2a3a9d699b993a4755f51bbf6970d32e80